### PR TITLE
[sc-3590] Resource fields edition, deletion and addition

### PR DESCRIPTION
**Pull request by mpellerin42**
- Navigation between fields
  - Display fields on the left side of profile view
  - Display profile by default when changing view
  - Improve nav block style
- Update and delete text, link and file fields
  - Display text field in a textarea
  - Ask for confirmation before deleting a field
  - update store in optimistic way
  - Add external label property on profile form's fields
- Field addition

### CODE REVIEW

File: apps/dashboard/src/app/resources/edit/edit-resource.service.ts [hunk 1]
```diff
@@ -54,8 +86,8 @@ export class EditResourceService {
         tap((kb) => this._resource.next(kb.getResourceFromData({ ...currentResource, ...partialResource }))),
       ),
     ]).pipe(
-      catchError((error) => {
-        this.toaster.error(error);
+      catchError(() => {
+        this.toaster.error('generic.error.oops');
         this._resource.next(currentResource);
         return of(null);
```
[the remark this answers is not in the document]
Comment (mpellerin42) [reply apps/dashboard/src/app/resources/edit/edit-resource.service.ts:92]: good catch, thanks!

File: apps/dashboard/src/app/resources/edit/add-field/add-field.component.html [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  </ng-container>
+
+  <ng-container *ngIf="fieldType.value === 'link'">
+    <div class="field-block">
+      <label
+        class="title-xxs"
+        for="link">
+        URL
+      </label>
+      <pa-input
```
Comment (operramon) [inline apps/dashboard/src/app/resources/edit/add-field/add-field.component.html:55]: As far as I remember, if the link URL doesn't include the URL schema (http/https), the backend can't process the link. In the link upload modal this issue is validated. Not sure if it should be validated here.

**Review by operramon — APPROVED** [review]

File: apps/dashboard/src/app/resources/edit/add-field/add-field.component.html (hunk 2 above)
Comment (mpellerin42) [reply apps/dashboard/src/app/resources/edit/add-field/add-field.component.html:55]: Good point, I didn't think about that. I'll add a validator

File: apps/dashboard/src/app/resources/edit/add-field/add-field.component.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { EditResourceService } from '../edit-resource.service';
+import { FIELD_TYPE, LinkField, Resource, TextField } from '@nuclia/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { BehaviorSubject, combineLatest, filter, map, Observable, of, Subject, switchMap, take, takeUntil } from 'rxjs';
+
+type TextFormat = 'PLAIN' | 'HTML' | 'RST' | 'MARKDOWN';
```
[the remark this answers is not in the document]
Comment (mpellerin42) [reply apps/dashboard/src/app/resources/edit/add-field/add-field.component.ts:7]: oh cool, I'll use it then, thanks

**Pull request merged**